wolfcrypt/src/tfm.c

```diff
@@ -2818,8 +2818,17 @@ int fp_exptmod(fp_int * G, fp_int * X, fp_int * P, fp_int * Y)
 
 #if defined(WOLFSSL_ESP32WROOM32_CRYPT_RSA_PRI) && \
    !defined(NO_WOLFSSL_ESP32WROOM32_CRYPT_RSA_PRI)
-   int x = fp_count_bits (X);
-#endif
+    #if defined(WOLFSSL_RSA_KEY_SIZE)
```

Comment:
Where is `WOLFSSL_RSA_KEY_SIZE` documented? Did you consider `RSA_MAX_SIZE`? Why is the logic not a >= 512? If this is ESP32 specific please use a macro that includes it.

Reply:
> Is the reason for these issues understood?  is the underlying bug/issue not understood yet?

No, not yet. I do not know exactly why the hardware-accelerated calc fails for certain RSA lengths. I wanted to at least get a fix in place before diving into the HW. It may take some time to fully resolve this. I chose to instead get other, higher visibility changes in place and backburner this for now.

> Where is WOLFSSL_RSA_KEY_SIZE documented? 

I neglected to add documentation on `WOLFSSL_RSA_KEY_SIZE`. I can add it if you are in favor of keeping this, or some other equivalent functionality.

The intent is for this value to be used for [explicit user RSA key sizes](https://github.com/gojimmypi/wolfssl/blob/9ef7d38e62d357334e1dd6b20831b1c91ca3e2c9/IDE/Espressif/ESP-IDF5/examples/wolfssl_server_6205/main/station_example_main.c#L261). We could put this in the default ESP32 `user_settings.h`. 

I didn't use `RSA_MAX_SIZE` as I wanted to define an explicit size, not a maximum size. Also, it appears other settings may [imply](https://github.com/wolfSSL/wolfssl/blob/6a89464176b1470f5c3432178b4510f5f46d2b1b/wolfssl/wolfcrypt/rsa.h#L127) a specific  `RSA_MAX_SIZE` value.

> Why is the logic not a >= 512

There's a specific problem for lengths of exactly 512 and a different problem for 2048 length.

>  If this is ESP32 specific please use a macro that includes it.

The TFM library only uses this gate when the `WOLFSSL_ESP32WROOM32_CRYPT_RSA_PRI` is enabled. Otherwise, I use it in the [demo app](https://github.com/gojimmypi/wolfssl/blob/ED25519_SHA2_fix/IDE/Espressif/ESP-IDF5/examples/wolfssl_server_6205/main/station_example_main.c) that I used when testing #6205.

Would you prefer an ESP32-specific name? It would seem best to have a macro for RSA lengths for the end user. The use of this macro in TFM does however, leave a bit to be desired. Once the HW math is fixed, the only use of this would be be the end user when calling something like `wc_MakeRsaKey`.

Reply:
Since this code is only for ESP32 I would remove `WOLFSSL_RSA_KEY_SIZE` logic and have this check always enabled and document the workaround for HW issue. The `WOLFSSL_RSA_KEY_SIZE` logic here is too limiting. RSA should support a range of key sizes.